py/torch_tensorrt/fx/converters/impl/convolution.py

```diff
@@ -0,0 +1,146 @@
+import numpy as np
+from typing import Any, Optional, Sequence, Union
+
+# @manual=//deeplearning/trt/python:py_tensorrt
+import tensorrt as trt
+import torch
+from torch.fx.node import Target
+
+from torch_tensorrt.fx.converters.converter_utils import (
+    SourceIR,
+    extend_attr_to_tuple,
+    get_dyn_range,
+    mark_as_int8_layer,
+    set_layer_name,
+    has_dynamic_shape,
+    to_numpy,
+    get_trt_tensor,
+)
+from torch_tensorrt.fx.converters.acc_ops_converters import (
+    acc_ops_unsqueeze,
+    acc_ops_squeeze,
+)
+
+from torch_tensorrt.fx.types import (
+    TRTNetwork,
+    TRTTensor,
+)
+
+
+def convNd(
+    network: TRTNetwork,
+    target: Union[Target, str],
+    source_ir: Optional[SourceIR],
+    name: str,
+    is_conv1d: bool,
+    input_val: TRTTensor,
+    weight: Union[TRTTensor, torch.Tensor],
+    bias: Optional[Union[TRTTensor, torch.Tensor]],
+    stride: Optional[Union[int, Sequence[int]]],
+    padding: Optional[Union[int, Sequence[int]]],
+    dilation: Optional[Union[int, Sequence[int]]],
+    groups: Optional[int],
+    scale: Optional[Union[torch.Tensor, float]] = None,
+    zero_point: Optional[Union[torch.Tensor, float]] = None,
+) -> TRTTensor:
+
+    if has_dynamic_shape(input_val.shape):
+        assert input_val.shape[1] != -1, "Channel dim can't be dynamic for convolution."
+
+    if is_conv1d:
+        # Apply an unsqueeze operation to transform the conv1d problem into conv2d
+        kwargs = {
+            "input": input_val,
+            "dim": -1,
+        }
+        input_val = acc_ops_unsqueeze(
+            network, target, tuple(), kwargs, name + "_unsqueeze"
+        )
+
+    # Process bias terms
+    if isinstance(bias, torch.Tensor):
+        # Transform the bias constant into a Numpy array
+        bias = to_numpy(bias)
+
+    elif isinstance(bias, TRTTensor):
+        bias = get_trt_tensor(network, bias, f"{name}_bias")
+
```

Comment:
Did not add an unsqueeze operation to bias term since the requirement in TRT for the bias term is that it must have number of elements equal to the number of output features of the convolution, so the same bias as is used for Conv1D would work for Conv2D, with the number of output features being fixed.

Reply:
just to clarify, you mean 1D or 2D conv? For 1D, we need bias to be unsqueezed.

Reply:
I initially meant for all conv layers since [this documentation](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html#iconvolutionlayer) seems to indicate we just need the number of elements in the bias Tensor to the be correct, and not necessarily the dimensions, but if bias needs to be unsqueezed for 1D, I can add that functionality back. I am wondering if the intended unsqueeze should be in the first dimension (`torch.unsqueeze(bias, 0)`) or the last dimension (`torch.unsqueeze(bias, -1)`)?

Reply:
I am wondering if TRT has done the broadcast internally since the unit test for 1D works good even though you did not unsqueeze it.

Reply:
I verified on a small sample that Conv1D with bias compiles + runs inference successfully without unsqueezing the bias term

Reply:
Should we have a unit test for this just to catch if TRT behavior changes?

Reply:
I could add one, but it would likely be very similar to this case, which interprets, builds, and runs inference with a Conv1D model with TRT both with and without bias:
https://github.com/pytorch/TensorRT/blob/cae6b7c694db1b6a5f287f9fc4e6c40ba08e1079/py/torch_tensorrt/fx/test/converters/acc_op/test_convolution.py#L8-L45